Content.Shared/_CM14/Comtech/Barbed/BarbedSystem.cs

```diff
@@ -0,0 +1,99 @@
+using Content.Shared._CM14.Comtech.Barbed.Components;
+using Content.Shared.Damage;
+using Content.Shared.Interaction;
+using Content.Shared.Weapons.Melee.Events;
+using Content.Shared.Popups;
+using Content.Shared.Stacks;
+using Content.Shared.DoAfter;
+using SharedToolSystem = Content.Shared.Tools.Systems.SharedToolSystem;
+using Content.Shared.Tools.Components;
+
+namespace Content.Shared._CM14.Barbed;
+
+public sealed class BarbedSystem : EntitySystem
+{
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedStackSystem _stacks = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedToolSystem _toolSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<BarbedComponent, AttackedEvent>(OnAttacked);
+        SubscribeLocalEvent<BarbedComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<BarbedComponent, BarbedDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<BarbedComponent, CutBarbedDoAfterEvent>(WireCutterOnDoAfter);
+    }
+    public void OnInteractUsing(EntityUid uid, BarbedComponent component, InteractUsingEvent args)
+    {
+        if (component.IsBarbed == false && HasComp<BarbedwireComponent>(args.Used))
+        {
+            var doAfterEventArgs = new DoAfterArgs(EntityManager, args.User, component.WireTime, new BarbedDoAfterEvent(), uid, used: args.Used)
+            {
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                NeedHand = true,
+            };
+            _popupSystem.PopupClient(Loc.GetString("barbed-wire-slot-wiring"), uid, args.User, PopupType.Small);
+            _doAfterSystem.TryStartDoAfter(doAfterEventArgs);
+            return;
+        }
+
+        if (component.IsBarbed == true && HasComp<BarbedwireComponent>(args.Used))
+        {
+            _popupSystem.PopupClient(Loc.GetString("barbed-wire-slot-insert-full"), uid, args.User, PopupType.Small);
+            return;
+        }
+
+        if (component.IsBarbed == true && TryComp<ToolComponent>(args.Used, out var tool))
+        {
+            if (_toolSystem.HasQuality(args.Used, "Cutting", tool))
+            {
+                _popupSystem.PopupClient(Loc.GetString("barbed-wire-cutting-action-begin"), uid, args.User, PopupType.Small);
```

Comment:
This and other popup strings should probably be specified on the component in the future as LocId data fields, but it's not too important at the moment, you can leave these as they are for this PR.

Reply:
I will add this as an issue then for the future!